src/algebra/star/pi.lean

```diff
@@ -58,4 +58,8 @@ lemma update_star [Π i, has_star (f i)] [decidable_eq I]
   function.update (star h) i (star a) = star (function.update h i a) :=
 funext $ λ j, (apply_update (λ i, star) h i a j).symm
 
+lemma star_sum_elim {I J α β : Type*} (x : I → α) (y : J → α) [has_star α] :
+  star (sum.elim x y)  = (sum.elim (star x) (star y)) :=
+by { ext x, cases x; simp }
+
```

Comment:
I think if we're going to have this, we should have lemmas about `-(sum.elim x y)` and `(sum.elim x y) + (sum.elim a b)` etc too; perhaps in `data.pi.algebra`.